mlflow/utils/requirements_utils.py

```diff
@@ -267,11 +267,16 @@ def _run_command(cmd, timeout_seconds, env=None):
             timer.cancel()
 
 
-def _get_installed_version(package, module=None):
+def _get_installed_version(package: str, module: Optional[str] = None) -> str:
     """
     Obtains the installed package version using `importlib_metadata.version`. If it fails, use
     `__import__(module or package).__version__`.
     """
+    if package == "mlflow":
+        # `importlib.metadata.version` may return an incorrect version of MLflow when it's
+        # installed in editable mode (e.g. `pip install -e .`).
+        return mlflow.__version__
```

Comment:
For example, `importlib.metadata.version("mlflow")` returns `2.20.3.dev0` on my machine. The last time I ran `pip install -e .`, it was that version.

Reply:
Oh, METADATA of the dist-info dir is not updated when the source code is updated..

Reply:
yep